src/snowflake/snowpark/mock/mock_connection.py

```diff
@@ -64,6 +71,63 @@ def _build_target_path(stage_location: str, dest_prefix: str = "") -> str:
 
 
 class MockServerConnection:
+    class TableRegistry:
```

Comment:
nit: I feel this can be a standalone class, no need to be nested in `MockServerConnection`

Reply:
For now I keep it here because it's a small class. In the future when we have table registry, UDF registry, etc, i.e. the mock datastore class I will move this to a new file. I left a TODO here 😅 .